### PR TITLE
Set user for rkt tasks

### PR DESCRIPTION
**Pull request by chelseakomlo**
Further testing is necessary for each case

### CODE REVIEW

File: client/driver/docker.go [hunk 1]
```diff
@@ -201,6 +201,7 @@ type DockerDriverConfig struct {
 	MacAddress       string              `mapstructure:"mac_address"`        // Pin mac address to container
 	SecurityOpt      []string            `mapstructure:"security_opt"`       // Flags to pass directly to security-opt
 	Devices          []DockerDevice      `mapstructure:"devices"`            // To allow mounting USB or other serial control devices
+	User             string              `json:"User,omitempty" yaml:"User,omitempty" toml:"User,omitempty"`
```
Comment (jippi) [inline client/driver/docker.go]: whats up with the yaml and toml here?

**Comment by chelseakomlo** [comment]
Testing suggestions are more than welcome.

File: client/driver/docker.go [hunk 2]
```diff
@@ -201,6 +201,7 @@ type DockerDriverConfig struct {
 	MacAddress       string              `mapstructure:"mac_address"`        // Pin mac address to container
 	SecurityOpt      []string            `mapstructure:"security_opt"`       // Flags to pass directly to security-opt
 	Devices          []DockerDevice      `mapstructure:"devices"`            // To allow mounting USB or other serial control devices
+	User             string              `mapstructure:user`
```
Comment (dadgar) [inline client/driver/docker.go]: Why add a new config field instead of using the task.User?

**Comment by chelseakomlo** [comment]
Changed this as we already expose this functionality for Docker, see https://www.nomadproject.io/docs/job-specification/task.html#user. 

Added the change to pass this through for rkt.

File: client/driver/rkt.go [hunk 3]
```diff
@@ -569,6 +569,11 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 
 	}
 
+	// If a user has been specified for the task, pass it through to the user
```
Comment (dadgar) [inline client/driver/rkt.go]: See if you can add tests to this and for docker

File: client/driver/rkt_test.go [hunk 4]
```diff
@@ -366,18 +368,19 @@ func TestRktDriverUser(t *testing.T) {
...
-	}
-	msg := "unknown user alice"
-	if !strings.Contains(err.Error(), msg) {
-		t.Fatalf("Expecting '%v' in '%v'", msg, err)
+	assert.Nil(err)
+	defer resp.Handle.Kill()
+
+	select {
+	case res := <-resp.Handle.WaitCh():
+		assert.False(res.Successful())
```
Comment (dadgar) [inline client/driver/rkt_test.go]: Can you assert it fails b/c the user

**Pull request merged**